src/components/organisms/Header.tsx

```diff
@@ -1,10 +1,93 @@
-import React, { ReactElement } from 'react'
+import React, { ReactElement, useEffect, useState } from 'react'
 import Menu from '../molecules/Menu'
 import styles from './Header.module.css'
+import AnnouncementBanner, {
+  AnnouncementAction
+} from '../molecules/AnnouncementBanner'
+import { useWeb3 } from '../../providers/Web3'
+import {
+  addCustomNetwork,
+  addOceanToWallet,
+  NetworkObject
+} from '../../utils/web3'
+import { getOceanConfig } from '../../utils/ocean'
+import { getProviderInfo, IProviderInfo } from 'web3modal'
+import { useOcean } from '../../providers/Ocean'
+import { useSiteMetadata } from '../../hooks/useSiteMetadata'
 
 export default function Header(): ReactElement {
+  const { web3Provider, networkId } = useWeb3()
+  const [providerInfo, setProviderInfo] = useState<IProviderInfo>()
+  const { config, connect } = useOcean()
+  const { warningPolygon, warningPolygonNetwork } = useSiteMetadata()
+
+  const network: NetworkObject = {
+    chainId: 137,
+    name: 'Matic Network',
+    urlList: [
+      'https://rpc-mainnet.matic.network',
+      'https://rpc-mainnet.maticvigil.com/'
+    ]
+  }
+  const [text, setText] = useState<string>(warningPolygonNetwork)
+  const [action, setAction] = useState<AnnouncementAction>()
+  const addCustomNetworkAction = {
```

Comment:
Feels like we should move out all that network-related logic into a dedicated component for only the network button, like `<ButtonNetwork>`. No need to tailor the `Header` for a specific announcement. Ideally, when we decide to remove an announcement we just remove the `AnnouncementBanner` import and are done, without having having to also remove all this network code.

Likewise, would be in favor of importing the `AnnouncementBanner` directly in `App.tsx` above the `Header`

Reply:
Yeah, also was thinking that maybe I should move all the logic to `AnnouncementBanner` component. Moving it to a new button component would mean that we would have to make the network-related logic in 2 places, for the button in the new component and somewhere else for the text. Maybe it's bette to keep them together.

Reply:
yup, let's start with that and then see if it makes sense abstracting further, so we have it out of `Header` at least. Also wasn't sure what exactly should be in the button when writing those use cases further down